### PR TITLE
feat(services-sessions): Add geoip parsing

### PR DESCRIPTION
**Pull request by saevarma**
[Notion task for geoip lookup](https://www.notion.so/Use-geoip-lite-for-location-lookup-of-ip-addresses-0300c6cafbeb456baecbc3ad14023b62?pvs=4)
[Notion taks for geoip update](https://www.notion.so/Update-geoip-lite-as-part-of-the-build-step-of-the-activities-service-15e86ac891154a93b1b25477fcf823b0?pvs=4)

## What

Adding geoip-lite lookup in sessions service for worker.
Adding `preBuildCommands` support in our custom build executor to run pre build step which updates the geoip data if running in CI.

@island-is/devops before merging we need to get island.is [sign up](https://www.maxmind.com/en/geolite2/signup?utm_source=kb&utm_medium=kb-link&utm_campaign=kb-create-account) on MaxMind webstie and [generate license key](https://support.maxmind.com/hc/en-us/articles/4407111582235-Generate-a-License-Key) and add as `GEOIP_LICENSE_KEY` environment variable for Github Actions.

## Why

To display location based on ip and update the geoip data.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/services/sessions/src/app/sessions/sessions.service.ts [hunk 1]
```diff
@@ -71,9 +72,15 @@ export class SessionsService {
     const device =
       browser || os ? `${browser}${browser && os ? ` (${os})` : os}` : undefined
 
+    const geoLocation = geoip.lookup(session.ip)
+    const ipLocation = geoLocation
+      ? `${geoLocation.city}, ${geoLocation.country}`
+      : undefined
```
Comment (eirikurn) [inline apps/services/sessions/src/app/sessions/sessions.service.ts:78]: Does this need to be more defensive or are both properties always returned?

File: tools/executors/node/schema.json [hunk 2]
```diff
@@ -161,6 +161,14 @@
       "type": "string",
       "description": "Name of the main output file. (defaults to `main.js`)",
       "default": "main.js"
+    },
+    "preBuildCommands": {
```
Comment (eirikurn) [inline tools/executors/node/schema.json:165]: Think we should use run-commands instead 👍

File: apps/services/sessions/src/app/sessions/sessions.service.ts (hunk 1 above)
Comment (saevarma) [reply apps/services/sessions/src/app/sessions/sessions.service.ts:78]: Looking at the source the country could be returned with empty city. So I will make this more defensive on the string formatting.

File: tools/executors/node/schema.json (hunk 2 above)
Comment (saevarma) [reply tools/executors/node/schema.json:165]: Agreed

File: apps/services/sessions/project.json [hunk 3]
```diff
@@ -4,6 +4,22 @@
   "projectType": "application",
   "targets": {
     "build": {
+      "executor": "@nrwl/workspace:run-commands",
+      "options": {
+        "outputPath": "dist/apps/services/sessions",
+        "commands": ["nx esbuild services-sessions"],
+        "parallel": false
```
Comment (eirikurn) [inline apps/services/sessions/project.json:11]: These should be fine to run in parallel ... right?

**Review by eirikurn — APPROVED** [review]
LGTM 👍

File: apps/services/sessions/project.json (hunk 3 above)
Comment (saevarma) [reply apps/services/sessions/project.json:11]: esbuild copies the package from node_modules, so we need to be sure that the data is updated before the build.

**Pull request merged**